lib/generators/revise_auth/model_generator.rb

```diff
@@ -0,0 +1,55 @@
+module ReviseAuth
+  module Generators
+    class ModelGenerator < Rails::Generators::NamedBase
+      include Rails::Generators::ResourceHelpers
+
+      desc "Generates a model for authentication, default User"
+
+      source_root File.expand_path("templates", __dir__)
+
+      argument :name, required: false, default: "User"
+      argument :attributes, type: :array, default: [], banner: "field:type field:type"
+
+      def generate_model
+        generate :model, name, *model_attributes
+      end
+
+      def add_revise_auth_model
+        inject_into_class model_path, class_name, " include ReviseAuth::Model\n"
+      end
+
+      def add_uniq_to_email_index
+        insert_into_file migration_path, after: "#{name.downcase.pluralize}, :email", force: true do
+          ", unique: true"
+        end
+      end
+
+      def done
+        readme "README" if behavior == :invoke
+      end
+
+      private
+
+      def migration_path
+        @migration_path ||= Dir.glob(Rails.root.join("db/migrate/*")).max_by { |f| File.mtime(f) }
+      end
+
+      def model_path
+        @model_path ||= File.join("app", "models", "#{file_path}.rb")
+      end
+
+      def model_attributes
+        [
+          "email:string:index",
+          "password_digest:string",
+          "confirmation_token:string",
+          "confirmed_at:datetime",
+          "confirmation_sent_at:datetime",
+          "unconfirmed_email:string"
+        ].concat(
+          attributes.map { "#{_1.name}:#{_1.type}" }
```

Comment:
This is a good idea. I think that we will lose any options and indexes though.

For example `account:belongs_to{polymorphic}` or `username:string:index` would be lost. 

I'm curious how Devise does this, but I bet we can override the `initialize` method and intercept the attributes before they're parsed. https://github.com/rails/rails/blob/main/railties/lib/rails/generators/named_base.rb#L17